fdbclient/BlobGranuleReader.actor.cpp

```diff
@@ -266,16 +268,28 @@ ACTOR Future<RangeResult> readBlobGranule(BlobGranuleChunkRef chunk,
 	// just to dependsOn instead of copy, to use a little extra memory footprint to help cpu?
 	state Arena arena;
 
+	// NOTE TO SELF: kept the increments to stats local to this fn 
+	// cuz didn't want to incur overhead of passing the stats object around
 	try {
 		state std::map<KeyRef, ValueRef> dataMap;
 
-		Future<Arena> readSnapshotFuture = chunk.snapshotFile.present()
-		                                       ? readSnapshotFile(bstore, chunk.snapshotFile.get(), keyRange, &dataMap)
-		                                       : Future<Arena>(Arena());
+		Future<Arena> readSnapshotFuture;
+		if (chunk.snapshotFile.present()) {
+			readSnapshotFuture = readSnapshotFile(bstore, chunk.snapshotFile.get(), keyRange, &dataMap);
+			if (stats.present()) {
+				++stats.get()->s3GetReqs;
```

Comment:
Kept the metrics updates local to this fn to avoid incurring overhead of passing the stats object around.